lib/actions/adapterActions.js

```diff
@@ -1097,6 +1109,42 @@ function closeSelectedAdapter(dispatch, getState) {
     });
 }
 
+export function selectedSerialPort(port) {
+    return dispatch => (
+        new Promise(() => {
+            dispatch(closeAdapter(() => {
+                getDeviceInfo(port.serialNumber)
+                    .then(deviceInfo => {
+                        dispatch(adapterGetLocalDeviceInfoAction(deviceInfo));
+                        logger.info(`Current device family is
+                            ${familyDefinitions[deviceInfo.family].split('_')[0].replace('NRF', 'nRF')}
+                            (${('0x0' + (deviceInfo.family).toString(16)).substring(-2)})`);
+                        logger.info(`Current device type is
+                            ${deviceTypeDefinitions[deviceInfo.deviceType].split('_')[0].replace('NRF', 'nRF')}
+                            (${('0x0' + (deviceInfo.deviceType).toString(16)).substring(-2)})`);
+                    })
+                    .then(() => {
+                        getFirmwareInfo(port.serialNumber)
+                            .then(versionInfo => {
+                                logger.info(`Current softdevice version is
+                                    ${versionInfo.sdBleApiVersion}
+                                    (${('0x0' + (versionInfo.sdBleApiVersion).toString(16)).substring(-2)})`);
```

Comment:
The SoftDevice version and baud rate are already logged with `logger.debug` in `openAdapter`. Debug messages are hidden from the log viewer, but end up in the log file. Could use `logger.info` in `openAdapter` if we want to display it in the log viewer.

Reply:
Good point.